src/CMakeLists.txt

```diff
@@ -53,40 +54,49 @@ target_include_directories(mgclient-static
 target_link_libraries(mgclient-static
         PRIVATE
         ${OPENSSL_LIBRARIES} project_options project_c_warnings)
+
 if(MGCLIENT_ON_WINDOWS)
     target_link_libraries(mgclient-static PUBLIC ws2_32)
 endif()
 
-add_library(mgclient-shared SHARED ${mgclient_src_files})
-set_target_properties(mgclient-shared PROPERTIES
-        OUTPUT_NAME mgclient
-        SOVERSION ${mgclient_SOVERSION}
-        C_VISIBILITY_PRESET hidden)
-target_include_directories(mgclient-shared
-        PRIVATE
-        "${PROJECT_SOURCE_DIR}/src"
-        PUBLIC
-        "${PROJECT_SOURCE_DIR}/include"
-        "${CMAKE_CURRENT_BINARY_DIR}"
-        "${OPENSSL_INCLUDE_DIR}")
-target_link_libraries(mgclient-shared PRIVATE ${OPENSSL_LIBRARIES})
+if(NOT EMSCRIPTEN)
+    add_library(mgclient-shared STATIC ${mgclient_src_files})
+    generate_export_header(mgclient-shared
+            BASE_NAME "mgclient"
+            EXPORT_FILE_NAME "mgclient-export.h")
+    set_target_properties(mgclient-shared PROPERTIES
+            OUTPUT_NAME mgclient
+            SOVERSION ${mgclient_SOVERSION}
+            C_VISIBILITY_PRESET hidden)
+    target_include_directories(mgclient-shared
+            PRIVATE
+            "${PROJECT_SOURCE_DIR}/src"
+            PUBLIC
+            "${PROJECT_SOURCE_DIR}/include"
+            "${CMAKE_CURRENT_BINARY_DIR}"
+            "${OPENSSL_INCLUDE_DIR}")
+    target_link_libraries(mgclient-shared PRIVATE ${OPENSSL_LIBRARIES})
+endif()
+
 if(MGCLIENT_ON_WINDOWS)
     target_link_libraries(mgclient-shared PUBLIC ws2_32)
 endif()
 
-generate_export_header(mgclient-shared
+generate_export_header(mgclient-static
```

Comment:
Why is this changed? We have to be careful with these things because maybe other clients depend on precisely this?

Reply:
We will have it static only for emcripten builds ;)